source/minio-cli/minio-kes-reference.rst

```diff
@@ -0,0 +1,759 @@
+======================================
+MinIO Key Encryption Service (``kes``)
+======================================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+.. mc:: kes
+
+Overview
+--------
+
+The MinIO Key Encryption Service (KES) is a stateless and distributed
+key-management system for high-performance applications. KES provides
+a bridge between applications running in bare-metal or orchestrated
+environments to centralised KMS solutions. 
+
+KES is designed for simplicity, scalability, and security. It requires 
+minimal configuration to enable full functionality and requires only
+basic familiarity with cryptography or key-management concepts.
+
+This page contains conceptual and reference information on using KES for
+performing cryptographic key operations and general data encryption/decryption.
+For documentation on deploying KES to support MinIO Server-Side Encryption
+(SSE-S3), see :ref:`minio-kes`.
+
+.. _minio-kes-installation:
+
+Installation
+------------
+
+.. include:: /includes/minio-kes-installation.rst
+
+.. _minio-kes-access-control:
+
+KES Access Control
+------------------
+
+KES uses mutual TLS authentication (mTLS) for performing both authentication and
+authorization of clients making requests to the KES server. 
+
+Authentication
+~~~~~~~~~~~~~~
+   
+Both the client and server present their x.509 certificate and
+corresponding private key to their peer. The server only accepts the
+connection if the client certificate is valid and authentic:
+
+- A "valid" certificate is well-formed and current (i.e. not expired). 
+   
+  mTLS also mTLS also requires the following `Extended Key Usage
+  <https://tools.ietf.org/html/rfc5280#section-4.2.1.12>`__ extensions for 
+  the client and server. Specifically:
+
+  - The client certificate *must* include Client Authentication
+     (``extendedKeyUsage = clientAuth``)
+  
+  - The server certificate *must* include Server Authentication
+     (``extendedKeyUsage = serverAuth``)
+
+- An "authentic" certificate is issued by a trusted Certificate Authority
+  (CA). Both client and server *must* include the peer certificate CA 
+  in their local system trust store.
+
+  You can start the KES server with the :mc-cmd-option:`kes server auth`
+  option to perform mTLS with untrusted certificates during testing or early
+  development. However, untrusted certificates present a security
```

Comment:
> However, untrusted certificates present a security
> vulnerability and may open the KES server to access by unknown parties.

This is actually not correct :wink: Even if you use "untrusted" - e.g self-signed certificates - KES remains completely secure. The difference between `--auth=off` and `--auth=on` is that in case of `--auth=off` any client can connect to the KES server while `--auth=on` enforces that the client has to use a "trusted" certificate to connect.
In any case, the KES server will apply policies to the client - regardless whether `--auth` is `on` or `off`. `--auth=on` is "just" a security enhancement / hardening.